mlc_llm/core.py

```diff
@@ -279,6 +281,13 @@ class BuildArgs:
             ),
         },
     )
+    use_presharded_weights: bool = field(
```

Comment:
Better to call it `use_presharded_weight` to align with `convert_weight_only`

Reply:
Hmm.  The current use of `weight` has been one of my most frequent typos, and I had hoped to avoid adding to that problem.  Since most models have more than a single parameter (citation needed), having the singular "weight" instead of "weights" always gives me a double-take.

What if we keep the plural `--use-presharded-weights`, and in a follow-up PR add an option for `--convert-weights-only` (plural)?  To maintain backwards compatibility, both `--convert-weights-only` and `--convert-weight-only` would be accepted by the argument parser, and would share the same the same argparse destination variable to be treated the same by the rest of the script.  That would allow consistency with the other arguments (presharded weights as a plural noun), while maintaining backwards compatibility.

Reply:
Sounds good, I also prefer a plural. If you can update to --convert-weights-only while keeping backwad compat, that would be the best!